lib/modules/manager/gomod/artifacts.ts

```diff
@@ -113,14 +116,23 @@ function addAuthFromHostRule(
 
 function getUpdateImportPathCmds(
   updatedDeps: PackageDependency[],
-  { constraints, newMajor }: UpdateArtifactsConfig
+  { constraints }: UpdateArtifactsConfig
 ): string[] {
   const updateImportCommands = updatedDeps
-    .map((dep) => dep.depName!)
-    .filter((x) => !x.startsWith('gopkg.in'))
+    .map(({ depName, newVersion }) => ({
+      depName: depName!,
+      newMajor: major(newVersion!),
```

Comment:
```suggestion
      newMajor: valid(newVersion) && major(newVersion),
```

Let's make this defensive as `major` can throw